### PR TITLE
Remove unnecessary array parenthesis

### PR DESCRIPTION
**Pull request by tkrotoff**
Draft: waiting for #1231 to be merged

Before:

```
/** A title */
fooBar: (string)[];
```

After:

```
/** A title */
fooBar: string[];
```

### CODE REVIEW

File: packages/openapi-typescript/src/utils.ts [hunk 1]
```diff
@@ -168,17 +168,22 @@ export function encodeRef(ref: string): string {
...
 
+/** if the type has & or | we should parenthesise it for safety */
+function parenthesise(type: string) {
+  return TS_UNION_INTERSECTION_RE.test(type) ? `(${type})` : type;
+}
+
 /** T[] */
 export function tsArrayOf(type: string): string {
-  return `(${type})[]`;
+  return `${parenthesise(type)}[]`;
```
Comment (tkrotoff) [inline packages/openapi-typescript/src/utils.ts:178]: The gist of it

**Review by drwpow — APPROVED** [review]
Awesome! Thank you for fixing this! All the generated TS seems to be valid—impressive!

**Comment by drwpow** [comment]
This and #1231 will be released in `6.3.6`. Thank you!

**Comment by tkrotoff** [comment]
> Awesome! Thank you for fixing this!

Thank you for this project, it works great ♥️

We use it in several projects at [Decathlon](https://decathlon.com)

**Comment by drwpow** [comment]
> We use it in several projects at [Decathlon](https://decathlon.com)

Whoa, what a cool site! Happy to have helped with a small bit of it 🙂

**Pull request merged**